components/Form.js

```diff
@@ -0,0 +1,218 @@
+import { useState } from 'react';
+import { nanoid } from 'nanoid';
+import { useRouter } from 'next/router';
+import { toast } from 'react-toastify';
+import Icon from '@mdi/react';
+import { mdiSend } from '@mdi/js';
+import {
+  FormControl,
+  TextField,
+  Select,
+  MenuItem,
+  TextareaAutosize,
+  FormLabel,
+  Radio,
+  RadioGroup,
+  FormControlLabel,
+  Stack,
+  Box,
+  Button,
+} from '@mui/material';
+
+import styled from 'styled-components';
+
+import { today } from '../utils/date';
+
+const StyledTextareaAutosize = styled(TextareaAutosize)`
+  padding: 8px;
+  margin-bottom: 5px;
+`;
+
+export default function Form() {
+  const router = useRouter();
+  const [form, setForm] = useState({
+    title: '',
+    location: '',
+    activity: '',
+    ageGroup: '',
+    description: '',
+    date: '',
+  });
+
+  function handleFormChange(event) {
+    const { name } = event.target;
+    let value = event.target.value;
+
+    if (value !== '' && name === 'title') {
+      value = value.trimStart();
+    }
+
+    setForm(prevState => ({
+      ...prevState,
+      [name]: value,
+    }));
```

Comment:
```suggestion
    setForm({
      ...form,
      [name]: value,
    });
```
Using a callback function here is not needed.

Reply:
i'll check this out